managed/src/main/java/com/yugabyte/yw/controllers/UniverseController.java

```diff
@@ -1762,4 +1781,42 @@ private UniverseDefinitionTaskParams bindFormDataToTaskParams(ObjectNode formDat
     formNode.remove(listType);
     return gflagMap;
   }
+
+  /**
+   * Throw an exception if the given provider has an AZ with
+   * KUBENAMESPACE in the config and the provdier has a cluster
+   * associated with it. Providers with namespace setting don't
+   * support multiple clusters.
+   * @param providerToCheck Provider object
+   */
+  private void checkK8sProviderAvailability(Provider providerToCheck) {
+    boolean isNamespaceSet = false;
+    for (Region r : Region.getByProvider(providerToCheck.uuid)) {
+      for (AvailabilityZone az : AvailabilityZone.getAZsForRegion(r.uuid)) {
+        if (az.getConfig().containsKey("KUBENAMESPACE")) {
+          isNamespaceSet = true;
+        }
+      }
+    }
+
+    if (isNamespaceSet) {
+      for (Universe allUniverseUUIDs : Universe.getAllUuids()) {
+        Universe u = Universe.get(allUniverseUUIDs.universeUUID);
+        List<Cluster> clusters = u.getUniverseDetails().getReadOnlyClusters();
+        clusters.add(u.getUniverseDetails().getPrimaryCluster());
+        for (Cluster c : clusters) {
+          UUID providerUUID = UUID.fromString(c.userIntent.provider);
+          if (providerUUID.equals(providerToCheck.uuid)) {
+            String msg = "Universe " + u.name + " (" + u.universeUUID
+              + ") already exists with provider "
+              + providerToCheck.name + " (" + providerToCheck.uuid
+              + "). Only one universe can be created with providers having KUBENAMESPACE set "
+              + "in the AZ config.";
```

Comment:
@iSignal What do you think about this error message? The resulting message is something like this:
```
Universe my-univ1 (xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxx) already exists with provider ocp-multi-az (xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxx). Only one universe can be created with providers having KUBENAMESPACE set in the AZ config.
```

Reply:
Had a chat with Sanketh offline, we are fine with this message for now.